Model/Order/Builder.php

```diff
@@ -110,7 +104,11 @@ public function build(Order $order)
                     $nostoOrder->setCreatedAt($orderCreatedDate);
                 }
             }
-            $nostoOrder->setPaymentProvider($order->getPayment()->getMethod());
+            if ($order->getPayment()) {
```

Comment:
Should we check the return type?

Reply:
Checked for `Magento\Sales\Model\Order\Payment`